### PR TITLE
Compatibility fixes for Isaac Sim 4.0.0

### PR DESCRIPTION
**Pull request by cgokmen**

### CODE REVIEW

File: omnigibson/utils/registry_utils.py [hunk 1]
```diff
@@ -351,7 +351,7 @@ def _load_state(self, state):
         # loading will be skipped.
         for obj in self.objects:
             if obj.name not in state:
-                log.warning(f"Object '{obj.name}' is not in the state dict to load from. Skip loading its state.")
+                # log.warning(f"Object '{obj.name}' is not in the state dict to load from. Skip loading its state.")
```
Comment (cgokmen) [inline omnigibson/utils/registry_utils.py:354]: Uncomment

File: omnigibson/utils/constants.py [hunk 2]
```diff
@@ -18,6 +18,14 @@
...
 
 
+class strWithCompare(str):
+    def __eq__(self, other):
+        return isinstance(other, str) and str.__eq__(self, other)
+
+
+MAGIC_DEFAULT = strWithCompare("default")
+
+
```
Comment (cgokmen) [inline omnigibson/utils/constants.py:28]: Remove all of this

File: omnigibson/simulator.py [hunk 3]
```diff
@@ -122,7 +122,7 @@ def _launch_app():
         ), "This version of OmniGibson supports Isaac Sim 2023.1.1 and above. Please update Isaac Sim."
 
     with launch_context(None):
-        app = lazy.omni.isaac.kit.SimulationApp(config_kwargs, experience=str(kit_file_target.resolve(strict=True)))
+        app = lazy.omni.isaac.kit.SimulationApp(config_kwargs)
```
Comment (cgokmen) [inline omnigibson/simulator.py:125]: Undo

File: omnigibson/simulator.py [hunk 4]
```diff
@@ -363,8 +363,8 @@ def __init__(
                     position=np.array(m.DEFAULT_VIEWER_CAMERA_POS),
                     orientation=np.array(m.DEFAULT_VIEWER_CAMERA_QUAT),
                 )
-                self.viewer_width = viewer_width
-                self.viewer_height = viewer_height
+                # self.viewer_width = viewer_width
+                # self.viewer_height = viewer_height
```
Comment (cgokmen) [inline omnigibson/simulator.py:367]: Figure out why uncommenting this breaks things.

File: omnigibson/simulator.py [hunk 5]
```diff
@@ -103,12 +103,12 @@ def _launch_app():
...
-    try:
-        shutil.copy(kit_file, kit_file_target)
-    except Exception as e:
-        raise e from ValueError("Failed to copy omnigibson.kit to Isaac Sim apps directory.")
+    # kit_file = Path(__file__).parent / "omnigibson.kit"
+    # kit_file_target = Path(os.environ["EXP_PATH"]) / "omnigibson.kit"
+    # try:
+    #     shutil.copy(kit_file, kit_file_target)
+    # except Exception as e:
+    #     raise e from ValueError("Failed to copy omnigibson.kit to Isaac Sim apps directory.")
```
Comment (cgokmen) [inline omnigibson/simulator.py:111]: We need to update the kit file. To do so, diff the current kit file with the existing kit file in Isaac Sim 2023.1.1's apps directory (its the python one in there) and get a .patch. Then apply the patch onto the new version's kit file. Include the patch in the repo here so it's easy to apply to future versions.

File: omnigibson/macros.py [hunk 6]
```diff
@@ -123,7 +123,7 @@ def determine_gm_path(default_path, env_var_name):
 gm.ENABLE_OBJECT_STATES = True
 
 # Whether to enable transition rules or not
-gm.ENABLE_TRANSITION_RULES = True
+gm.ENABLE_TRANSITION_RULES = False
```
Comment (cgokmen) [inline omnigibson/macros.py:126]: Undo

File: omnigibson/controllers/controller_base.py [hunk 7]
```diff
@@ -125,13 +126,13 @@ def __init__(
         self._command_input_transform = None
 
         # Standardize command input / output limits to be (min_array, max_array)
-        command_input_limits = (-1.0, 1.0) if command_input_limits == "default" else command_input_limits
+        command_input_limits = (-1.0, 1.0) if MAGIC_DEFAULT == command_input_limits else command_input_limits
```
Comment (cgokmen) [inline omnigibson/controllers/controller_base.py:129]: Remove all the references to this MAGIC_DEFAULT. Instead, put this check in: `type(command_input_limits) == str and command_input_limits == "default"`. Do this for all of the examples below.

File: omnigibson/macros.py (hunk 6 above)
Comment (hang-yin) [reply omnigibson/macros.py:126]: We haven't fixed transition rules on `og-develop` though. Do we want to keep this False for now?

File: omnigibson/simulator.py (hunk 4 above)
Comment (hang-yin) [reply omnigibson/simulator.py:371]: This is super confusing but I just figured this out: essentially the problem here is the new replicator api does not support having two render products at the same time. Our code used to hold the old render product and the new render product at the same time to detach and attach all our annotators. Now, we will have to detach all the annotators from the old render product, destroy the old one, create a new one, and attach the annotators.

File: omnigibson/macros.py (hunk 6 above)
Comment (cgokmen) [reply omnigibson/macros.py:126]: Why is this showing as a diff then? Did we not merge og-develop here?

File: omnigibson/sensors/vision_sensor.py [hunk 8]
```diff
@@ -596,14 +596,16 @@ def image_height(self, height):
...
 
-        old_render_product.destroy()
-        self._render_product = new_render_product
+        self._render_product.destroy()
+        self._render_product = lazy.omni.replicator.core.create.render_product(
+            self.prim_path, (width, height), force_new=True
+        )
+
+        for annotator in self._annotators.values():
+            annotator.attach([self._render_product])
```
Comment (cgokmen) [inline omnigibson/sensors/vision_sensor.py:608]: What's happening here? Are we undoing Josiah's change?

Comment (hang-yin) [reply omnigibson/sensors/vision_sensor.py:608]: Not really, we are just changing the order of doing things here

Comment (hang-yin) [reply omnigibson/sensors/vision_sensor.py:608]: Please check this comment: https://github.com/StanfordVL/OmniGibson/pull/776#discussion_r1653721379

**Review by hang-yin — APPROVED** [review]

**Pull request merged**